src/cryptenroll/cryptenroll-tpm2.c

```diff
@@ -74,55 +75,61 @@ static int get_pin(char **ret_pin_str, TPM2Flags *ret_flags) {
         assert(ret_pin_str);
         assert(ret_flags);
 
-        r = getenv_steal_erase("NEWPIN", &pin_str);
-        if (r < 0)
-                return log_error_errno(r, "Failed to acquire PIN from environment: %m");
+        // try to get the pin from systemd-creds
+        r = read_credential("NEWPIN", (void**) &pin_str, NULL);
```

Comment:
don't call this NEWPIN, but something like tpm2.pin or so

Reply:
@bluca I was thinking today morning: does make sense to be sensitive to the env var NEWPIN? Wouldn't be better to just switch it to the cred?

Reply:
env vars are useful for debugging